build/source/engine/groundwatr.f90

```diff
@@ -419,117 +328,81 @@ subroutine computeBaseflow(&
- end do  ! looping through soil layers
+ end do  ! end looping through soil layers
 
  ! set un-used portions of the vectors to zero
- if(ixSaturation>1)then
+ if (ixSaturation>1) then
   zActive(1:ixSaturation-1) = 0._rkind
   trTotal(1:ixSaturation-1) = 0._rkind
   trSoil(1:ixSaturation-1)  = 0._rkind
  end if
 
- ! compute the outflow from each layer (m3 s-1)
- mLayerColumnOutflow(1:nSoil) = trSoil(1:nSoil)*tan_slope*contourLength
+ mLayerColumnOutflow(1:nSoil) = trSoil(1:nSoil)*tan_slope*contourLength ! compute the outflow from each layer (m3 s-1)
 
  ! compute total column inflow and total column outflow (m s-1)
  totalColumnInflow  = sum(mLayerColumnInflow(1:nSoil))/HRUarea
  totalColumnOutflow = sum(mLayerColumnOutflow(1:nSoil))/HRUarea
 
- ! compute the available storage (m)
- availStorage = sum(mLayerDepth(1:nSoil)*(theta_sat - (mLayerVolFracLiq(1:nSoil)+mLayerVolFracIce(1:nSoil))) )
+ availStorage = sum(mLayerDepth(1:nSoil)*(theta_sat - (mLayerVolFracLiq(1:nSoil)+mLayerVolFracIce(1:nSoil)))) ! compute the available storage (m)
 
  ! compute the smoothing function (-)
- if(availStorage < xMinEval)then
-  ! (compute the logistic function)
+ if (availStorage < xMinEval) then
+  ! compute the logistic function
   expF = exp((availStorage - xCenter)/xWidth)
   logF = 1._rkind / (1._rkind + expF)
   ! (compute the derivative in the logistic function w.r.t. volumetric liquid water content in each soil layer)
-  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2._rkind
+  dLogFunc_dLiq(1:nSoil) = mLayerDepth(1:nSoil)*(expF/xWidth)/(1._rkind + expF)**2_i4b !!integer exponent
  else
   logF             = 0._rkind
   dLogFunc_dLiq(:) = 0._rkind
  end if
 
- ! compute the exfiltartion (m s-1)
+ ! compute the exfiltration (m s-1)
  if(totalColumnInflow > totalColumnOutflow .and. logF > tiny(1._rkind))then
   scalarExfiltration = logF*(totalColumnInflow - totalColumnOutflow)  ! m s-1
-  !write(*,'(a,1x,10(f30.20,1x))') 'scalarExfiltration = ', scalarExfiltration
  else
   scalarExfiltration = 0._rkind
  end if
 
- ! check
- !write(*,'(a,1x,10(f30.20,1x))') 'zActive(1), soilDepth, availStorage, logF, scalarExfiltration = ', &
- !                                 zActive(1), soilDepth, availStorage, logF, scalarExfiltration
- !if(scalarExfiltration > tiny(1.0_rkind))  pause 'exfiltrating'
-
- ! compute the baseflow in each layer (m s-1)
- mLayerBaseflow(1:nSoil) = (mLayerColumnOutflow(1:nSoil) - mLayerColumnInflow(1:nSoil))/HRUarea
+ mLayerBaseflow(1:nSoil) = (mLayerColumnOutflow(1:nSoil) - mLayerColumnInflow(1:nSoil))/HRUarea ! compute the baseflow in each layer (m s-1)
 
- ! compute the total baseflow
- qbTotal = sum(mLayerBaseflow)
+ qbTotal = sum(mLayerBaseflow) ! compute the total baseflow
 
  ! add exfiltration to the baseflow flux at the top layer
  mLayerBaseflow(1)      = mLayerBaseflow(1) + scalarExfiltration
  mLayerColumnOutflow(1) = mLayerColumnOutflow(1) + scalarExfiltration*HRUarea
 
- ! test
```

Comment:
@martynpclark Can you also give an opinion on whether it's worth keeping (formal) debugging statements like these?